collated/functional/yeggasd.md

```diff
@@ -0,0 +1,1231 @@
+
+        } catch (BadPaddingException e) {
+            handleBadPaddingException(inputBytes, e);
+        } catch (IllegalBlockSizeException e) {
+            logger.info("Warning: Text already in plain text.");
+        }
+    }
+
+    /**
+     * Hashes the DEFAULT_PASSWORD to meet the required length for AES.
+     */
+    public static byte[] hashPassword(String password) {
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-1");
+            try {
+                md.update(password.getBytes("UTF-8"));
+            } catch (UnsupportedEncodingException e) {
+                logger.severe("UTF-8 not supported, using default but may not be able to decrypt in other computer.");
+                md.update(password.getBytes());
+            }
+            byte[] hashedPassword = md.digest();
+            return Arrays.copyOf(hashedPassword, 16);
+        } catch (NoSuchAlgorithmException nsae) {
+            throw new AssertionError("SHA-1 should exist");
+        }
+    }
+
+    /**
+     * Decrypt the given file with the current and if it fails the previous password
+     * if password is null, will not try to decrypt
+     * @param file Points to the file to be decrypted
+     * @param password Used to decrypt file
+     * @throws IOException if file cannot be opened
+     * @throws WrongPasswordException if password used is wrong
+     */
+    public static void decryptFile (File file, Password password) throws IOException, WrongPasswordException {
+        if (password.getPassword() != null) {
+            try {
+                decrypt(file, password.getPassword());
+            } catch (WrongPasswordException e) {
+                logger.info("Current Password don't work, trying previous password.");
+                decrypt(file, password.getPrevPassword());
+            }
+        }
+    }
+
+    /**
+     * Encrypt the given file with the current
+     * if password is null, will not try to encrypt
+     * @param file Points to the file to be decrypted
+     * @param password Used to decrypt file
+     * @throws IOException if file cannot be opened
+     * @throws WrongPasswordException if password used is wrong
+     */
+    public static void encryptFile (File file, Password password) throws IOException, WrongPasswordException {
+        if (password.getPassword() != null) {
+            encrypt(file, password.getPassword());
+        }
+    }
+    /**
+     * Generates a key.
+     */
+    private static Key createKey(byte[] password) {
+        return new SecretKeySpec(password, "AES");
+    }
+
+    /**
+     * Handles {@code BadPaddingException} by determining whether it is plain text or other case
+     * @param inputBytes Input data that caused this
+     * @param e Contains the exception details to throw with WrongPasswordException
+     * @throws WrongPasswordException if it is wrong password
+     */
+    private static void  handleBadPaddingException(byte[] inputBytes, BadPaddingException e)
```

Comment:
Using my eye power it seems like there are two spaces between the `void` and `handleBadPaddingException`...please check if this is the case :P